Mathlib/Data/Polynomial/RingDivision.lean

```diff
@@ -402,6 +415,24 @@ theorem comp_X_add_C_eq_zero_iff {p : R[X]} (t : R) :
 theorem comp_X_add_C_ne_zero_iff {p : R[X]} (t : R) :
     p.comp (X + C t) ≠ 0 ↔ p ≠ 0 := Iff.not <| comp_X_add_C_eq_zero_iff t
 
+theorem Monic.C_dvd_iff_isUnit {p : R[X]} (hp : Monic p) {a : R} :
+    C a ∣ p ↔ IsUnit a :=
+  ⟨fun h => isUnit_iff_dvd_one.mpr <|
+      hp.coeff_natDegree ▸ (C_dvd_iff_dvd_coeff _ _).mp h p.natDegree,
+   fun ha => IsUnit.dvd (ha.map C)⟩
```

Comment:
I think these lemmas should go into a CommSemiring section rather than CommRing, and
```suggestion
   (·.map C).dvd⟩
```
should work here (if you like it)